src/ansible_grid.c

```diff
@@ -1777,20 +1787,18 @@ void handler_KriaGridKey(s32 data) {
 					}
 				}
 			}
-			else if(k_mode == mPattern) {
-				if(y == 0) {
-					if(!meta) {
-						if(cue) {
-							cue_pat_next = x+1;
-						}
-						else {
-							change_pattern(x);
-						}
+			else if ((k_views[0].mode == mPattern && y == 0 ) || (k_views[1].mode == mPattern && y == 8)) {
```

Comment:
this changes the logic which could potentially cause issues - previously it would never reach the below `else if(view_tuning)` if it was in pattern mode and now it's possible if `y` is not 0 or 8.

Reply:
Hmm, taking a closer look, the original logic doesn't seem right either -- the `view_clock`/`view_config`/`view_tuning` state should take priority over any "normal" mode logic, as it does later in the grid key handler. This fast-press section seems like it ought to be ordered the same way, with the `view_` clauses ahead of the mode checks.

Reply:
Taking a closer look at this, I think the original logic is incorrect -- `view_tuning` should take precedence over the "normal" modes. With the original code, it looks like if you were in pattern mode and entered the tuning view, you wouldn't be able to use row 0, those grid presses would still be interpreted as pattern changes. I think the right behavior here is to move the `view_tuning` clause above the pattern mode checks, the same as it is in the non-quick-change logic later in the handler.

Reply:
yeah that makes sense. do you want to tackle that change or leave it for a separate fix?

Reply:
already fixed it in c8123c2! sorry, should have been more explicit.